lale/lib/lale/group_by.py

```diff
@@ -12,16 +12,76 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import ast
+
+import numpy as np
+import pandas as pd
+
 import lale.docstrings
 import lale.operators
 
+try:
+    from pyspark.sql.dataframe import DataFrame as spark_df
+
+    spark_installed = True
+
+except ImportError:
+    spark_installed = False
+
+
+def _is_pandas_df(df):
```

Comment:
These functions `_is_pandas_df`, `_is_spark_df`, `_is_ast_subscript`, `_is_ast_attribute` seem to occur in multiple operators. Aggregate might have a different requirement for the first two, but for other places, wondering if we should move them to utils. This can be one of the clean-up activities once we write an end-to-end example.

Reply:
As discussed, will include these changes in next PR.